sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/NonStreamingOrderByDocumentQueryExecutionContext.java

```diff
@@ -0,0 +1,268 @@
+import com.azure.cosmos.implementation.query.orderbyquery.OrderbyRowComparer;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.models.FeedResponse;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.azure.cosmos.models.SqlQuerySpec;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class NonStreamingOrderByDocumentQueryExecutionContext
+    extends ParallelDocumentQueryExecutionContextBase<Document> {
+
+    private final static
+    ImplementationBridgeHelpers.CosmosDiagnosticsHelper.CosmosDiagnosticsAccessor diagnosticsAccessor =
+        ImplementationBridgeHelpers.CosmosDiagnosticsHelper.getCosmosDiagnosticsAccessor();
+
+    private static final ImplementationBridgeHelpers.FeedResponseHelper.FeedResponseAccessor feedResponseAccessor =
+        ImplementationBridgeHelpers.FeedResponseHelper.getFeedResponseAccessor();
+
+    private final static String FormatPlaceHolder = "{documentdb-formattableorderbyquery-filter}";
+    private final static String True = "true";
+
+    private final OrderbyRowComparer<Document> consumeComparer;
+    private final RequestChargeTracker tracker;
+    private final ConcurrentMap<String, QueryMetrics> queryMetricMap;
+    private final Collection<ClientSideRequestStatistics> clientSideRequestStatistics;
+    private Flux<OrderByRowResult<Document>> orderByObservable;
+
+    private int maxPageSizePerPartition;
+
+    public NonStreamingOrderByDocumentQueryExecutionContext(
+        DiagnosticsClientContext diagnosticsClientContext,
+        IDocumentQueryClient client,
+        ResourceType resourceTypeEnum,
+        SqlQuerySpec query,
+        CosmosQueryRequestOptions cosmosQueryRequestOptions,
+        String resourceLink,
+        String rewrittenQuery,
+        OrderbyRowComparer<Document> consumeComparer,
+        UUID correlatedActivityId,
+        boolean hasSelectValue,
+        final AtomicBoolean isQueryCancelledOnTimeout,
+        int maxPageSizePerPartition) {
+        super(diagnosticsClientContext, client, resourceTypeEnum, Document.class, query, cosmosQueryRequestOptions,
+            resourceLink, rewrittenQuery, correlatedActivityId, hasSelectValue, isQueryCancelledOnTimeout);
+        this.consumeComparer = consumeComparer;
+        this.tracker = new RequestChargeTracker();
+        this.queryMetricMap = new ConcurrentHashMap<>();
+        this.clientSideRequestStatistics = ConcurrentHashMap.newKeySet();
+        this.maxPageSizePerPartition = maxPageSizePerPartition;
```

Comment:
where the maxPageSizePerPartition comes from?

Reply:
Removed, missed by mistake in the previous iteration.